packages/sampler/stories/default/ViewManager.js

```diff
@@ -0,0 +1,135 @@
+
+const ViewManagerConfig = mergeComponentMetadata('ViewManager', ViewManagerBase, ViewManager);
+
+ViewManagerConfig.defaultProps = {
+	duration: 300,
+	enteringDelay: 0
+};
+
+const prop = {
+	arranger: ['SlideBottomArranger', 'SlideLeftArranger', 'SlideRightArranger', 'SlideTopArranger', 'SlideArranger', 'CustomArranger (FadeAndSlideArranger)'],
+	direction: ['bottom', 'left', 'right', 'top']
+};
+
+const arrangers = {
+	'SlideBottomArranger': SlideBottomArranger,
+	'SlideLeftArranger': SlideLeftArranger,
+	'SlideRightArranger': SlideRightArranger,
+	'SlideTopArranger': SlideTopArranger
+};
+
+const views = new Array(10).fill().map((i, index) => {
+	return {
+		title: `Item ${index + 1}`,
+		content: `This is View ${index + 1}`
+	};
+});
+
+const ViewManagerLayout = (props) => {
+	const [selected, setSelected] = useState(0);
+	const handleChangeView = useCallback((ev) => {
+		setSelected(ev.selected);
+	}, [setSelected]);
+
+	return (
+		<Layout>
+			<Cell component={Scroller} size="7%">
+				<Group
+					className={css.group}
+					childComponent={Item}
+					itemProps={{className: css.navItem}}
+					onSelect={handleChangeView}
+					select="radio"
+				>
+					{views.map((view) => view.title)}
+				</Group>
+			</Cell>
+			<Cell
+				className={css.viewport}
+				component={ViewManager}
+				index={selected}
+				{...props}
+			>
+				{views.map((view, i) => (
+					<div className={css.box} key={i}>
+						{view.content}
+					</div>
+				))}
+			</Cell>
+		</Layout>
+	);
+};
+
+export default {
+	title: 'UI/ViewManager',
+	component: 'ViewManager'
+};
+
+export const _ViewManager = () => {
+	const arrangerType = select('arranger', prop.arranger, ViewManagerConfig, prop.arranger[0]);
+	let arranger;
+
+	if (arrangerType === 'SlideArranger') {
+		const amount = number('amount', ViewManagerConfig, {range: true, min: 0, max: 100}, 100);
+		const direction = select('direction', prop.direction, ViewManagerConfig, prop.direction[0]);
+		arranger = SlideArranger({amount, direction});
+	} else if (arrangerType === 'CustomArranger (FadeAndSlideArranger)') {
+		// The following arranger is from sandstone/internal/Panels/Arrangers.
+		const quadInOut = 'cubic-bezier(0.455, 0.030, 0.515, 0.955)';
+		const animationOptions = {easing: quadInOut};
+		const FadeAndSlideArranger = {
+			enter: (config) => {
+				return arrange(config, [
+					{transform: `translateX(${config.rtl ? '-' : ''}100%)`, opacity: 0, offset: 0},
+					{opacity: 0, offset: 0.5},
+					{transform: 'none', opacity: 1, offset: 1}
+				], animationOptions);
+			},
+			leave: (config) => {
+				return arrange(config, [
+					{transform: 'none', opacity: 1, offset: 0},
+					{opacity: 0, offset: 0.5},
+					{transform: `translateX(${config.rtl ? '' : '-'}100%)`, opacity: 0, offset: 1}
+				], animationOptions);
+			}
+		};
+		arranger = FadeAndSlideArranger;
+	} else {
+		arranger = arrangers[arrangerType];
+	}
+
+	return (
+		<ViewManagerLayout
```

Comment:
Why there is no `start` and `end` prop?

In fact the reason I need Viewmanager sample is fot this two props. it releated with WRN-853.

Reply:
That was because those props can't be set by fixed default value.
I've added those props with an empty initial state.

Reply:
This is not what I meant, and I cannot any  behavior about `start` and `end` in this current sample.
Since `start` and `end` are values that depend on the current index, So they cannot be received as single knobs.
If we need knobs, I think we can only delta value based on the index.

Please check the behavior using the code below.
```
				start={selected - 1}
				end={selected + 1}
				index={selected}
```

```
.viewport {
	// height: 42px;
.box {
	// height: 100%;
```
I thought the reason I expected this issue to take long was because of these props.

Reply:
I've changed how `start` and `end` work in this sample.
It works with some selected values among the range that depend on the current index.
Also, the height of the viewport is dependent on how many views show.